### PR TITLE
Tests: Add logging in loops in p2p_sendhears.py

### PR DESCRIPTION
**Pull request by ccdle12**
PR for #12453 

New contributor looking at mainly the 'good first issues'. 

From my understanding of the issue:
 * Track the iteration level of loop when test fails in Travis CI

Further clarification on the issue would be greatly appreciated to narrow down what can be fixed or updated.

### CODE REVIEW

File: test/functional/p2p_sendheaders.py [hunk 1]
```diff
@@ -380,6 +380,8 @@ def test_nonnull_locators(self, test_node, inv_node):
         # PART 3.  Headers announcements can stop after large reorg, and resume after
         # getheaders or inv from peer.
         for j in range(2):
+            self.log.info("Part 3: iteration level: {}".format(j))
+
```
Comment (MarcoFalke) [inline test/functional/p2p_sendheaders.py]: I'd slightly prefer if the comments in the loops down there were transformed into debug statements. (Where we check for `i == ...` and `j == ..`

File: test/functional/p2p_sendheaders.py [hunk 2]
```diff
@@ -406,6 +408,8 @@ def test_nonnull_locators(self, test_node, inv_node):
             test_node.wait_for_block(new_block_hashes[-1])
 
             for i in range(3):
+                self.log.debug("Part 3: i == {}".format(i))
+
                 # Mine another block, still should get only an inv
```
Comment (MarcoFalke) [inline test/functional/p2p_sendheaders.py]: What I meant was to transfer those comments from `# ...` to `self.log.debug(...`

File: test/functional/p2p_sendheaders.py [hunk 3]
```diff
@@ -381,7 +381,7 @@ def test_nonnull_locators(self, test_node, inv_node):
         # getheaders or inv from peer.
         for j in range(2):
             # First try mining a reorg that can propagate with header announcement
-            new_block_hashes = self.mine_reorg(length=7)
+            new_block_hashes = self.mine_reorg(length=7)
```
Comment (jnewbery) [inline test/functional/p2p_sendheaders.py]: nit: you've added trailing whitespace here.

File: test/functional/p2p_sendheaders.py [hunk 4]
```diff
@@ -412,11 +412,13 @@ def test_nonnull_locators(self, test_node, inv_node):
                 test_node.check_last_announcement(inv=[tip], headers=[])
                 if i == 0:
                     # Just get the data -- shouldn't cause headers announcements to resume
```
Comment (jnewbery) [inline test/functional/p2p_sendheaders.py]: You can remove this comment now that you have a log saying the same thing!

Same for comments lower down.

**Review by jnewbery — COMMENTED** [review]
Thanks for your first contribution @ccdle12!

I'm not sure if you've dropped a commit, but this PR isn't doing what the comments suggest.

A couple of nits inline.

File: test/functional/p2p_sendheaders.py (hunk 4 above)
Comment (ccdle12) [reply test/functional/p2p_sendheaders.py]: Thanks @jnewbery, appreciate the patience for a newbie, have pushed a new commit remove the comments above the debug prints.

File: src/test/.vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+    "files.associations": {
+        "*.ipp": "cpp"
+    }
+}
```
Comment (MarcoFalke) [inline src/test/.vscode/settings.json]: Unwanted file?

File: test/functional/p2p_sendheaders.py [hunk 6]
```diff
@@ -411,21 +411,18 @@ def test_nonnull_locators(self, test_node, inv_node):
                 inv_node.check_last_announcement(inv=[tip], headers=[])
                 test_node.check_last_announcement(inv=[tip], headers=[])
                 if i == 0:
-                    # Just get the data -- shouldn't cause headers announcements to resume
```
Comment (MarcoFalke) [inline test/functional/p2p_sendheaders.py]: Please remove this as well.

**Review by maflcko — COMMENTED** [review]
Please squash your commits according to https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits

**Comment by jnewbery** [comment]
As it stands, this doesn't seem to be that useful. It'll print out the same log many times over as it iterates over `i` and `j`.

**Comment by MarcoFalke** [comment]
utACK 8dd547d82b70678c19bee110950d962d4e62ed5a. Seems like a tiny improvement over what it was before.

**Pull request merged**